src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +46,383 @@ ActorInit En_Takaraya_InitVars = {
     (ActorFunc)EnTakaraya_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80ADFB20[] = {
+void* D_80ADFB00[4] = { 0x06006F58, 0x06007358, 0x06007758, 0x06007358 };
+
+void* D_80ADFB10[4] = { 0x06007B58, 0x06007F58, 0x06007758, 0x06007F58 };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_U8(targetMode, TARGET_MODE_6, ICHAIN_CONTINUE),
     ICHAIN_F32(targetArrowOffset, 1000, ICHAIN_STOP),
 };
 
-#endif
+u32 D_80ADFB28 = 0;
+
+u16 D_80ADFB2C[PLAYER_FORM_MAX] = {
+    1901, // PLAYER_FORM_FIERCE_DEITY
+    1902, // PLAYER_FORM_GORON
+    1903, // PLAYER_FORM_ZORA
+    1900, // PLAYER_FORM_DEKU
+    1901, // PLAYER_FORM_HUMAN
+};
+
+u8 D_80ADFB38[PLAYER_FORM_MAX][2] = {
+    { 5, 5 },   // PLAYER_FORM_FIERCE_DEITY
+    { 12, 5 },  // PLAYER_FORM_GORON
+    { 4, 4 },   // PLAYER_FORM_ZORA
+    { 42, 41 }, // PLAYER_FORM_DEKU
+    { 5, 5 },   // PLAYER_FORM_HUMAN
+};
+
+u16 D_80ADFB44[PLAYER_FORM_MAX] = {
+    1909, // PLAYER_FORM_FIERCE_DEITY
+    1910, // PLAYER_FORM_GORON
+    1911, // PLAYER_FORM_ZORA
+    1908, // PLAYER_FORM_DEKU
+    1909, // PLAYER_FORM_HUMAN
+};
+
+u16 D_80ADFB50[PLAYER_FORM_MAX] = {
+    1905, // PLAYER_FORM_FIERCE_DEITY
+    1906, // PLAYER_FORM_GORON
+    1907, // PLAYER_FORM_ZORA
+    1904, // PLAYER_FORM_DEKU
+    1905, // PLAYER_FORM_HUMAN
+};
+
+void EnTakaraya_Init(Actor* thisx, PlayState* play) {
+    EnTakaraya* this = THIS;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, -60.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &object_bg_Skel_008FC8, &object_bg_Anim_009890, this->jointTable,
+                       this->morphTable, OBJECT_BG_1_LIMB_MAX);
+    this->switchFlag = EN_TAKARAYA_GET_SWITCH_FLAG(thisx);
+    thisx->params = EN_TAKARAYA_GET_FF(thisx);
```

Comment:
Ah this is actually probably a clear command not a get, the goal here was to grab the switch flag from above first, then clear it off so that the treasure flag could be read later as just this->actor.params without any additional bit logic.

```suggestion
    this->switchFlag = EN_TAKARAYA_GET_SWITCH_FLAG(thisx);
    thisx->params &= 0xFF;
```